src/python/pants/backend/python/rules/python_create_binary.py

```diff
@@ -59,11 +59,8 @@ def create(cls, tgt: Target) -> "PythonBinaryFields":
     if fields.entry_point.value is not None:
         entry_point = fields.entry_point.value
     else:
-        # TODO: rework determine_source_files.py to work with the Target API. It should take the
-        #  Sources AsyncField as input, rather than TargetAdaptor.
-        sources_result = await Get[SourcesResult](SourcesRequest, fields.sources.request)
         stripped_sources = await Get[SourceRootStrippedSources](
-            StripSnapshotRequest(sources_result.snapshot)
+            StripSourcesFieldRequest(fields.sources)
```

Comment:
The followup queued up for after this PR will replace this to use `await Get[SourceFiles](AllSourceFilesRequest([fields.sources], strip_source_roots=True)`.